LandSandBoat/server
# [Magic] Banish Spell Functionality

Implements Banish SDT Debuffs when used vs undead mobs. Implements Afflatus Misery functionality to Banish spells. Implements WHM Merit bonuses for Banish Spells.
Implements gear bonuses for "Potency of Banish vs Undead"



**_I affirm:_**

## What does this pull request do?

Implements various missing functionality for Banish series spells.
See link for more details on specific SDT Values.
https://www.bg-wiki.com/ffxi/Category:Banish_Spells

1. Adds missing SDT debuff vs undead type mobs.
2. Adds Afflatus Misery bonus for Banish series spells.
3. Adds WHM Banish Merit bonuses.
4. Adds gear modifiers for "Potency of Banish vs Undead" (BANISH_POTENCY)

**_Things that still need to be added:_** Duration resistance. Duration resistance should be independent of spell damage resists. (Currently unsure what all influences this)

Things to note:
- The SDT reduction can not go above 100% (Will never give bonus damage, only reduces resistance value like Tomahawk).
- Gear modifiers are calculated multiplicatively. Example: Banish 1 = -50% reduction to SDT resistance. Gear Mod = 10. Final Value would be -55%, not -60%.



## Steps to test these changes

1. Find Skeleton.
2. Use `!getmod pierce_sdt` on mob. Note value as 500 (50% damage reduction).
3. Cast banish spell on mob.
4. Use `!getmod pierce_sdt` on mob. See that the value has changed.
5. Wait for effect to wear off, check once more and see that the value goes back to 500.
6. Use `!additem clerics_mitts` , equip item. Cast Banish again.
7. Use `!getmod pierce_sdt` on mob. See that the value has changed from step 4.

8. Banish Merits give +2 MATT to Banish spells and +2 seconds to the duration.

9. Afflatus Misery: Spell damage increases based on last damage instance.
See: https://wiki.ffo.jp/html/1963.html
